CourseNGRX/src/app/courses/home/home.component.ts

```diff
@@ -24,17 +80,63 @@ export class HomeComponent implements OnInit {
     this.deleteModal = new window.bootstrap.Modal(
       document.getElementById('deleteModal')
     );
+
+    this.store.dispatch(invokeCoursesApi());
   }
 
   openCreateModal() {
     this.createModal.show();
   }
 
-  openEditModal() {
+  openEditModal(id: number) {
+    let fetchFormData$ = this.route.paramMap.pipe(
+      switchMap(() => {
+        return this.store.pipe(select(selectCourseById(id)));
+      })
+    );
+
+    fetchFormData$.subscribe((data) => {
+      if (data) {
+        this.EditCourseForm = { ...data };
+      } else {
+        this.router.navigateByUrl('');
+      }
+    });
+
     this.editModal.show();
   }
 
-  openDeleteModal() {
+  update() {
+    this.store.dispatch(
+      invokeUpdateCourseApi({ payload: { ...this.EditCourseForm } })
+    );
+
+    let appstate$ = this.appState.pipe(select(selectAppState));
+    appstate$.subscribe((data) => {
+      if (data.apiStatus === 'sucess') {
```

Comment:
Explica-me o racional deste subscribe sff

Reply:
eu fiz um store para verificar o estado (esse this.appState).  O subscribe esta ai apenas para retornar o valor emitido pelo observable appstate$